internal/fabric/rpc.go

```diff
@@ -249,10 +250,10 @@ func (w *rpcWrapper) QueryChainInfo(channelId, signer string) (*fab.BlockchainIn
 }
 
 // The returned registration must be closed when done
-func (w *rpcWrapper) SubscribeEvent(channelId, signer string, since uint64) (fab.Registration, <-chan *fab.BlockEvent, *event.Client, error) {
-	client, err := w.getChannelClient(channelId, signer)
+func (w *rpcWrapper) SubscribeEvent(subInfo *eventsapi.SubscriptionInfo, since uint64) (fab.Registration, <-chan *fab.BlockEvent, <-chan *fab.CCEvent, *event.Client, error) {
```

Comment:
This is a lot of return values. To avoid accidentally getting the order of them mixed up, should we return a struct here with named fields? Maybe my suggestion is too "object oriented" but just a thought.

Reply:
it's a fair question and something that came across my mind as well, decided to go with this given that each value is of a different type so apart from the cosmetics there's no danger with returning the values out of order by accident